pkg/v7/chartconfig/chartconfig.go

```diff
@@ -0,0 +1,69 @@
+package chartconfig
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/tenantcluster"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Config represents the configuration used to create a new chartconfig service.
+type Config struct {
+	Logger micrologger.Logger
+	Tenant tenantcluster.Interface
+
+	ProjectName string
+}
+
+// Service provides shared functionality for managing chartconfigs.
+type Service struct {
+	logger micrologger.Logger
+	tenant tenantcluster.Interface
+
+	projectName string
+}
+
+// New creates a new chartconfig service.
+func New(config Config) (*Service, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.Tenant == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Guest must not be empty", config)
+	}
+
+	if config.ProjectName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ProjectName must not be empty", config)
+	}
+
+	s := &Service{
+		logger: config.Logger,
+		tenant: config.Tenant,
+
+		projectName: config.ProjectName,
+	}
+
+	return s, nil
+}
+
+func getChartConfigByName(list []*v1alpha1.ChartConfig, name string) (*v1alpha1.ChartConfig, error) {
+	for _, l := range list {
+		if l.Name == name {
+			return l, nil
+		}
+	}
+
+	return nil, microerror.Mask(notFoundError)
+}
+
+func (s *Service) getTenantK8sClient(ctx context.Context, clusterConfig ClusterConfig) (kubernetes.Interface, error) {
```

Comment:
I would stick with `newTenantK8sClient` because it seems to create something new.

Reply:
Thanks that's better. Done.